### PR TITLE
Reset beatmap mods on exiting song select screen

### LINKED ISSUE #1961: Songs in DT bugs in Songselection / Main Menu
When you have a **Song running** and are **in Song Selection** and **enable DT**, when you **go to Main Menu**, your **Song will remain being DoubleTime**.

**Re-entering Song Selection resets DoubleTime mod** and the Song will play on normal speed.

I do not know whether this is a bug, but I think it is due to re-entering disabling DT. If yes, to fix there is two options:
1. Disable DT when leaving Song Selection
2. Let DT on when entering Song Selection

**Read bold for tl;dr**

### PR DESCRIPTION
**Pull request by UselessToucan**
Resolves #1961 
There are no null checks because I did not see any null values here during manual testing(both with and without beatmaps imported).

### CODE REVIEW

**Comment by ThePyrotechnic** [comment]
I think that, while this fix works, the problem to be solved is whether mod selections should persist when returning to the menu. In stable they currently do, with the only difference being that DT/NC and HT don't actually change the speed/pitch.

**Comment by UselessToucan** [comment]
@ThePyrotechnic 
> the problem to be solved is whether mod selections should persist when returning to the menu.

In my opinion, this thing is not related to #1961 .
Mods are not saved because we create new song select screen whenever returning to main menu.

**Comment by ThePyrotechnic** [comment]
@UselessToucan I believe that it is related because if the decision was made that mod selections should persist through returning to the main menu, then this PR would have to be reverted. Even if it were decided that the speed and pitch should be reset and then set again upon reentering the the selection screen, a different solution would be required which involves holding onto the state of the mods.

**Comment by UselessToucan** [comment]
I did not know that it was decided to save mod selection. :-(
Will investigate situation tomorrow and update/rename this PR.

**Comment by LittleEndu** [comment]
Regardless, people who want to use osu as jukebox would appreciate if track speed is reset when returning to main menu. I don't see why we can't have both, that is, DT/NC/HT/DC would just reapply when going back to song select.

File: osu.Game/Screens/Menu/MainMenu.cs [hunk 1]
```diff
@@ -161,6 +161,8 @@ protected override void OnResuming(Screen last)
         {
             base.OnResuming(last);
 
+            Beatmap.Value.Track.ResetSpeedAdjustments();
```
Comment (smoogipoo) [inline osu.Game/Screens/Menu/MainMenu.cs]: Rather than this, `PlaySongSelect` should set Beatmap.Value.Mods to an empty list.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Comment by peppy** [comment]
Not sure resetting mods is a logical thing to do. It's not what I'd expect as a user.

**Comment by ThePyrotechnic** [comment]
What @LittleEndu said about the jukebox is definitely important, but I think that if someone deliberately enables those mods it would be strange if the modulation stopped in the menu but then returned when back in the selection screen. 
Assuming that mods should persist, which I think is a good idea (not that my opinion should have much weight), it would probably make more sense to leave the effect on in the main menu.

**Comment by peppy** [comment]
If mods are only available to change from song select inwards, they should not be applied at a higher level.

**Comment by peppy** [comment]
Closing as #1983 resolves this in a better way.

**Pull request closed without merging**